airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/JdbcStateManager.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.source.jdbc;
+
+import com.google.common.collect.Lists;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.source.jdbc.models.JdbcState;
+import io.airbyte.integrations.source.jdbc.models.JdbcStreamState;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class JdbcStateManager {
+
+  private final Map<String, CursorInfo> map;
+
+  public JdbcStateManager(JdbcState serialized) {
+    map = serialized.getStreams()
+        .stream()
+        .peek(s -> {
+          if (s.getCursorField().size() == 0) {
+            throw new IllegalStateException("No cursor field specified for stream attempting to do incremental.");
+          } else if (s.getCursorField().size() > 1) {
+            throw new IllegalStateException("JdbcSource does not support composite cursor fields.");
+          }
+        })
+        .collect(Collectors.toMap(JdbcStreamState::getStreamName, s -> new CursorInfo(s.getCursorField().get(0), s.getCursor())));
+  }
+
+  private Optional<CursorInfo> getCursorInfo(String streamName) {
+    return Optional.ofNullable(map.get(streamName));
+  }
+
+  public Optional<String> getOriginalCursorField(String streamName) {
```

Comment:
this is a little goofy as opposed to just returning the cursor info. i didn't want to have to worry about defensive copy issues though.